### DISCUSSION TITLE
Is the QR control language becoming Turing complete?

### DISCUSSION THREAD

**Discussion by andrerav** [gopro/labs#245] [General]
After peeking at the latest commits to docs/control/actions/README.md I am wondering if the QR control language is starting to become Turing complete? With the latest upcoming changes there will be conditionals, loops and variable assignments available for the programmer. Some limitations exist, such as program length and nesting depth of conditionals, of course. What do you think?

**Comment by dnewman-gpsw** [comment]
It's getting there, however maintaining backwards compatibility with existing QR commands and the tiny code size, I doubt if will ever be Turing complete.  That is not the goal, all the new power is to help users craft custom triggers, with a side benefit it is help in batch testing Labs. If there a missing coding feature you would like, or improved, let me know.
